chia/full_node/block_store.py

```diff
@@ -294,6 +297,39 @@ async def get_full_blocks_at(self, heights: List[uint32]) -> List[FullBlock]:
                 ret.append(self.maybe_decompress(row[0]))
             return ret
 
+    async def get_generators_at(self, heights: List[uint32]) -> List[SerializedProgram]:
+        assert self.db_wrapper.db_version == 2
+
+        if len(heights) == 0:
+            return []
+
+        generators: Dict[uint32, SerializedProgram] = {}
+        heights_db = tuple(heights)
+        formatted_str = (
+            f"SELECT block, height from full_blocks "
+            f'WHERE in_main_chain=1 AND height in ({"?," * (len(heights_db) - 1)}?)'
+        )
+        async with self.db.execute(formatted_str, heights_db) as cursor:
+            async for row in cursor:
+                if self.db_wrapper.db_version == 2:
```

Comment:
guaranteed to be 2, are we doing the if just for consistency?

Reply:
I would say the parsing is the bulk of the speedup, especially on SSD. I'll run some measurements

Reply:
oh, that response ended up in the wrong thread somehow. Regarding this thread; that's a good point. I wasn't considering that this was a v2-only function to begin with. I'll simplify this